packages/polaris-viz/src/components/FunnelChartNext/Chart.tsx

```diff
@@ -0,0 +1,289 @@
+                  index={index}
+                  isLast={isLast}
+                  onMouseEnter={(index) => setTooltipIndex(index)}
+                  onMouseLeave={() => setTooltipIndex(null)}
+                  shouldApplyScaling={shouldApplyScaling}
+                  x={x}
+                >
+                  {!isLast && (
+                    <FunnelChartConnector
+                      drawableHeight={drawableHeight}
+                      height={drawableHeight}
+                      index={index}
+                      nextX={
+                        (xScale(nextPoint?.key as string) ?? 0) - LINE_OFFSET
+                      }
+                      nextY={drawableHeight - nextBarHeight}
+                      startX={x + barWidth + GAP}
+                      startY={drawableHeight - barHeight}
+                    />
+                  )}
+                </FunnelChartSegment>
+                {index > 0 && (
+                  <rect
+                    y={PERCENTAGE_SUMMARY_HEIGHT}
+                    x={x - (LINE_OFFSET - LINE_WIDTH)}
+                    width={LINE_WIDTH}
+                    height={drawableHeight - PERCENTAGE_SUMMARY_HEIGHT}
+                    fill={`url(#${lineGradientId})`}
+                  />
+                )}
+              </g>
+            </Fragment>
+          );
+        })}
+
+        <TooltipWithPortal>{getTooltipMarkup()}</TooltipWithPortal>
+      </g>
+    </ChartElements.Svg>
+  );
+
+  function getTooltipMarkup() {
+    if (tooltipIndex == null) {
+      return null;
+    }
+
+    const tooltipHeight =
+      tooltipIndex === dataSeries.length - 1
+        ? SHORT_TOOLTIP_HEIGHT
+        : TOOLTIP_HEIGHT;
+
+    const activeDataSeries = dataSeries[tooltipIndex];
+
+    if (activeDataSeries == null) {
+      return null;
+    }
+
+    const xPosition = getXPosition();
+    const yPosition = getYPosition();
+
+    return (
+      <FunnelTooltip x={xPosition} y={yPosition}>
+        <Tooltip
+          activeIndex={tooltipIndex}
+          dataSeries={dataSeries}
+          isLast={tooltipIndex === dataSeries.length - 1}
+          tooltipLabels={tooltipLabels}
+          yAxisOptions={yAxisOptions}
+        />
+      </FunnelTooltip>
+    );
+
+    function getXPosition() {
+      if (tooltipIndex === 0) {
+        // Push the tooltip beside the bar
+        return chartX + barWidth + 10;
+      }
+
+      // Center the tooltip over the bar
+      const xOffset = (barWidth - TOOLTIP_WIDTH) / 2;
+      return chartX + (xScale(activeDataSeries.key.toString()) ?? 0) + xOffset;
+    }
+
+    function getYPosition() {
+      const barHeight = getBarHeight(activeDataSeries.value ?? 0);
+      const yPosition = chartY + drawableHeight - barHeight;
+
+      if (tooltipIndex === 0) {
+        return yPosition;
+      }
+
+      return yPosition - tooltipHeight;
+    }
+  }
+
+  function formatPercentage(value: number) {
+    return `${yAxisOptions.labelFormatter(isNaN(value) ? 0 : value)}%`;
```

Comment:
I think that would make sense but let's check with UX. 

If the handling is for dividing by 0, can we specifically check for that? I want to avoid covering up some other NaN case unintentionally.

Reply:
Pablo is out this week but I will make a note to follow up. I'v made some changes to explicitly check for division by 0